libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -71,22 +73,206 @@ public void Initialize(ref ArgSlice[] buffer, int count)
+        }
+
         /// <summary>
         /// Initialize the parse state with a given set of arguments
         /// </summary>
-        /// <param name="buffer">Reference to arguments buffer (necessary for keeping buffer object rooted)</param>
         /// <param name="args">Set of arguments to initialize buffer with</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void InitializeWithArguments(ref ArgSlice[] buffer, params ArgSlice[] args)
+        public void InitializeWithArguments(ArgSlice[] args)
         {
-            Initialize(ref buffer, args.Length);
+            Initialize(args.Length);
 
             for (var i = 0; i < args.Length; i++)
             {
                 buffer[i] = args[i];
             }
         }
 
+        /// <summary>
+        /// Set argument at a specific index
+        /// </summary>
+        /// <param name="i">Index of buffer at which to set argument</param>
+        /// <param name="arg">Argument to set</param>
+        public void SetArgument(int i, ArgSlice arg)
+        {
+            Debug.Assert(i < Count);
+            buffer[i] = arg;
+        }
+
+        /// <summary>
+        /// Set arguments starting at a specific index
+        /// </summary>
+        /// <param name="i">Index of buffer at which to start setting arguments</param>
+        /// <param name="args">Arguments to set</param>
+        public void SetArguments(int i, params ArgSlice[] args)
+        {
+            Debug.Assert(i + args.Length - 1 < Count);
+            for (var j = 0; j < args.Length; j++)
+            {
+                buffer[i + j] = args[j];
+            }
+        }
+
+        /// <summary>
+        /// Get serialized length of parse state when arguments are only
+        /// serialized starting at a specified index
+        /// </summary>
+        /// <param name="startIdx">Index from which arguments are serialized</param>
+        /// <returns>The serialized length</returns>
+        public int GetSerializedLength(int startIdx)
+        {
+            var serializedLength = sizeof(int);
+            var argCount = Count - startIdx;
+
+            if (argCount > 0)
+            {
+                Debug.Assert(startIdx < Count);
+
+                for (var i = 0; i < argCount; i++)
+                {
+                    serializedLength += buffer[startIdx + i].SpanByte.TotalSize;
+                }
+            }
+
+            return serializedLength;
+        }
+
+        /// <summary>
+        /// Serialize parse state to memory buffer
+        /// when arguments are only serialized starting at a specified index
+        /// </summary>
+        /// <param name="dest">The memory buffer to serialize into (of size at least SerializedLength(startIdx) bytes)</param>
+        /// <param name="startIdx">Index from which arguments are serialized</param>
+        /// <returns>Total serialized bytes</returns>
+        public int CopyTo(byte* dest, int startIdx)
```

Comment:
Looks like this needs to be `public int CopyTo(byte* dest, int startIdx, int count)`.